docs/developer/index.rst

```diff
@@ -24,9 +24,11 @@ Before you submit a pull request to Bodhi, please ensure that it meets these cri
   ``flake8`` utility to automatically check your code.
 * Add an entry to `docs/user/release_notes.rst`_ for any changes you make that should be in release
   notes.
-* Make sure your commits are atomic. Each commit should focus on one improvement or bug fix. If you
-  need to build upon changes that are related but aren't atomic, feel free to send more than one
-  commit in the same pull request.
+* Make sure your commits are atomic. With only rare exceptions, each improvement or bug fix should
+  have exactly one commit. This makes it much easier to peruse the git history to find out what
+  changes occurred, and is particularly valuable when commits need to be cherry picked. If you need
```

Comment:
`what changes occurred` could also match changes that were made during a pull request for fixing remarks, which I think is not what we want.
How about `find out which changes relate to a feature or bugfix implementation`?